ldn-access-request.md

```diff
@@ -0,0 +1,103 @@
+# Requesting Access with Linked Data Notifications
+
+Status: draft
+
+
+## Abstract
+This proposal describes an exchange of Linked Data Notifications with a specific shape
+for the purpose of requesting access to a resource.
+
+
+## Actors
+- **Resource server:**
+  server on which the resource is located
+- **Requesting agent:**
+  agent requesting access to the resource
+- **Processing agent:**
+  agent accepting or rejecting the access request
+
+
+## Request flow
+
+1. The requesting agent assumes it does not have access to a given resource `R`.
+   For instance, it might have received a `403` in response to `GET` request,
+   or a `404` for a resource it knows exists.
+
+2. The requesting agent discovers an _access request inbox_ of `R`,
+   of from an agent that has access to `R`.
+   This might have been communicated out-of-band,
+   or can be linked with a `tdb:accessRequestInbox` relation,
+   or with a generic `ldp:inbox` relation.
+
+3. The requesting agent sends an _access request_
+   conforming to the shape described below
+   using the [Linked Data Notifications](https://www.w3.org/TR/ldn/) Protocol.
+
+4. The processing agent _optionally_ (depending on the request)
+   sends an _accept_ or _reject_ notification to an inbox of the requesting agent.
+
+5. The requesting agent retries accessing the resource.
+
+
+## Request access shape
+
+_This example needs to be formalized into a SHACL and/or ShEx shape._
+
+In this example,
+an agent with WebID `https://user/#me`
+is requesting Read and Write permissions for itself
+on a resource `https://resources/documents/x`.
+
+It submits this message to an inbox of `https://resources/documents/x`
+(or to an inbox of an administrators for this resource)
+in which it requests appending a permission block
+to the resource's ACL document at `https://resources/documents/x.acl`.
+
+
+```turtle
+@prefix acl: <http://www.w3.org/ns/auth/acl#>.
+@prefix as: <https://www.w3.org/ns/activitystreams#>.
+@prefix tbd: <http://example.org/to-be-determined#>.
+
+[
```

Comment:
Suggest use a URI to identify the request like `<#this>` so it can be linked to in any response